src/token/common/erc2981/erc2981.cairo

```diff
@@ -0,0 +1,166 @@
+        ) -> (ContractAddress, u256) {
+            let royalty_info = self.token_royalty_info.read(token_id);
+
+            let (royalty_receiver, royalty_fraction) = if royalty_info.receiver.is_zero() {
+                let default_info = self.default_royalty_info.read();
+                (default_info.receiver, default_info.royalty_fraction)
+            } else {
+                (royalty_info.receiver, royalty_info.royalty_fraction)
+            };
+
+            let royalty_amount: u256 = (sale_price * royalty_fraction) / self._fee_denominator();
+
+            (royalty_receiver, royalty_amount)
+        }
+    }
+
+
+    //
+    // Internal
+    //
+    #[generate_trait]
+    pub impl InternalImpl<
+        TContractState,
+        +HasComponent<TContractState>,
+        impl SRC5: SRC5Component::HasComponent<TContractState>,
+        +Drop<TContractState>
+    > of InternalTrait<TContractState> {
+        /// Initializes the contract by setting default royalty.
+        fn initializer(
+            ref self: ComponentState<TContractState>,
+            default_receiver: ContractAddress,
+            default_royalty_fraction: u256
+        ) {
+            let mut src5_component = get_dep_component_mut!(ref self, SRC5);
+            src5_component.register_interface(IERC2981_ID);
+
+            self._set_default_royalty(default_receiver, default_royalty_fraction)
+        }
+
+
+        /// The denominator with which to interpret the fee set in {_set_token_royalty} and
+        /// {_set_default_royalty} as a fraction of the sale price.
+        /// Defaults to 10000 so fees are expressed in basis points
+        fn _fee_denominator(self: @ComponentState<TContractState>) -> u256 {
+            default_fee_denominator
+        }
+
+        /// Returns the royalty information that all ids in this contract will default to.
+        fn _default_royalty(
+            self: @ComponentState<TContractState>
+        ) -> (ContractAddress, u256, u256) {
+            let royalty_info = self.default_royalty_info.read();
+            (royalty_info.receiver, royalty_info.royalty_fraction, self._fee_denominator())
+        }
+
+        /// Sets the royalty information that all ids in this contract will default to.
+        ///
+        /// Requirements:
+        ///
+        /// - `receiver` cannot be the zero address.
+        /// - `fee_numerator` cannot be greater than the fee denominator.
+        fn _set_default_royalty(
+            ref self: ComponentState<TContractState>,
+            receiver: ContractAddress,
+            fee_numerator: u256,
+        ) {
+            let denominator = self._fee_denominator();
+            assert!(fee_numerator <= denominator, "Invalid default royalty");
+            assert!(receiver.is_non_zero(), "Invalid default royalty receiver");
+            self
+                .default_royalty_info
+                .write(RoyaltyInfo { receiver, royalty_fraction: fee_numerator })
+        }
+
```

Comment:
This implementation lacks `_deleteDefaultRoyalty` function, which is present in the Solidity implementation. Imo, it makes sense to add it for consistency

Reply:
Could you please confirm if the following behavior is expected after `_deleteDefaultRoyalty` is called ?:
- Default Royalty is 0% and receiver address is address(0)

Reply:
Yes, it is the expected one